From 404da358937b2b496dfec549c8fd86966a22d5ab Mon Sep 17 00:00:00 2001
From: Anurag Kumar Vulisha <anurag.kumar.vulisha@xilinx.com>
Date: Thu, 23 Jan 2020 19:50:58 +0530
Subject: [PATCH 1010/1852] usb: xhci: don't clean shared_hcd when otg is
 enabled

commit c9240cf5ded047c29b7263c344bad5bfdfa3a957 from
https://github.com/Xilinx/linux-xlnx.git

Since xhci->shared_hcd contains all the required information
that is needed by otg driver, don't clear the xhci->shared_hcd
during xhci_stop() when otg is enabled

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Anurag Kumar Vulisha <anurag.kumar.vulisha@xilinx.com>
Signed-off-by: Piyush Mehta <piyush.mehta@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/usb/host/xhci.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/usb/host/xhci.c b/drivers/usb/host/xhci.c
index 2d39d464d9fb..97ab8f1a04dd 100644
--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -727,6 +727,11 @@ static void xhci_stop(struct usb_hcd *hcd)
 
 	/* Only halt host and free memory after both hcds are removed */
 	if (!usb_hcd_is_primary_hcd(hcd)) {
+		/* Remove shared_hcd if no otg ports are present */
+		if (!hcd->self.otg_port) {
+			/* usb core will free this hcd shortly, unset pointer */
+			xhci->shared_hcd = NULL;
+		}
 		mutex_unlock(&xhci->mutex);
 		return;
 	}
-- 
2.31.1

